app/models/decidim/participations/participation.rb

```diff
@@ -23,11 +23,24 @@ class Participation < Participations::ApplicationRecord
 
       geocoded_by :address, http_headers: ->(participation) { { "Referer" => participation.feature.organization.host } }
 
+      # upstream moderation => MOA dashboard
+      scope :unmoderate, ->(current_feature) { current_feature_participations(current_feature).joins(:moderation).merge(Moderation.where(upstream_moderation: "unmoderate")) }
```

Comment:
so these all scopes are the same structure as I can tell, it just specifies the `upstream_moderation`

1. you could use an internal scope with an argument that would be the `upstream_moderation` values.

```
scope :filter_upstream_moderation,  lambda { |array|
  where("upstream_moderation in (?)", array)
}

Participation.filter_upstream_moderation(['refused'])
```

2. Could we benefit of having an array or enum of upstream_moderation values?

Reply:
To edit later